src/passkit.cs

```diff
@@ -2645,4 +2754,62 @@ interface PKDeferredPaymentRequest {
 		[DesignatedInitializer]
 		NativeHandle Constructor (string paymentDescription, PKDeferredPaymentSummaryItem deferredBilling, NSUrl managementUrl);
 	}
+
+	[NoWatch, NoTV, NoMac, iOS (17, 0), NoMacCatalyst]
+	[BaseType (typeof (UIView))]
+	[DisableDefaultCtor]
+	interface PKPayLaterView {
+		// [DesignatedInitializer]
+		[Export ("initWithAmount:currencyCode:")]
+		NativeHandle Constructor (NSDecimalNumber amount, string currencyCode);
```

Comment:
As is, this fails intro with: 
```
[FAIL] PassKit.PKPayLaterView should re-expose UIView::.ctor(CoreGraphics.CGRect)
	[FAIL] DesignatedInitializer :   1 potential errors found in 835 designated initializer validated
  Expected: 0
  But was:  1

		  at Introspection.ApiCtorInitTest.DesignatedInitializer () [0x0015d] in /Users/tjlambert/Microsoft/Attributes/xamarin-macios/tests/introspection/ApiCtorInitTest.cs:351 
```
but when I uncomment the [DesignatedInitializer], then it fails xtro with:
```
!extra-designated-initializer! PKPayLaterView::initWithAmount:currencyCode: is incorrectly decorated with an [DesignatedInitializer] attribute
```

Is this one that needs a test ignored?

Reply:
This is a long shot but the API was a bit interesting and I think the context of it might help..Turns out there are actually 4 display styles offered for the view, most being more generic and not requiring the price/currency params:

<img width="822" alt="image" src="https://github.com/xamarin/xamarin-macios/assets/110641567/6ace8a5a-2243-4121-b03d-2af7b1cf9fc0">

Ref: https://developer.apple.com/videos/play/wwdc2023/10114 

So I wonder if the fix would be to remove the `[DisableDefaultCtor]`? That way both the API could get access to UIView base type's ctor and `[DesignatedInitializer]` wouldn't be necessary?

Reply:
The header for `PKPayLaterView` has this:

```objective-c
- (instancetype)init NS_UNAVAILABLE;
+ (instancetype)new NS_UNAVAILABLE;
- (instancetype)initWithFrame:(CGRect)frame NS_UNAVAILABLE;
- (instancetype)initWithCoder:(NSCoder *)coder NS_UNAVAILABLE;
```

which means `[DisableDefaeultCtor]` is correct, and that intro needs a manual exception.

Reply:
Ah I see! Thank you!